back/engines/commercial/smart_groups/app/models/smart_groups/patches/project.rb

```diff
@@ -0,0 +1,15 @@
+module SmartGroups
+  module Patches
+    module Project
+      extend ActiveSupport::Concern
+
+      included do
+        before_destroy :remove_rules
+      end
+
+      def remove_rules
+        SmartGroups::RulesService.new.filter_by_value_references(id).each(&:destroy!)
```

Comment:
I see how this removes rules that relate to a specific project, but I am wondering if smart groups that use this rule will still work. I assume the smart group membership will be recalculated based on any remaining rules. Correct? And what if the smart group used only the rule that is removed? Is the smart group then destroyed / removed also? (This is not necessarily a problem, I am just wondering about the consequences)

Reply:
A far as I know, we don't deal with that and you would get a smart group with no rules. Very good point. I think this means everyone belongs to that smart group. To make things even worse, we probably shouldn't delete the smart group if it's used for project access conditions (e.g. only people who participated in project A can participate in project B). Perhaps we could consider logging this as tech dept?